File: Data Structures & Algorithms/sliding-window-maximum/submission-6.swift
class Solution {
    func maxSlidingWindow(_ nums: [Int], _ k: Int) -> [Int] {

        let n = nums.count

        var res:[Int] = []


        for i in 0...n-k {

           
            var maxi = Int.min
            for j in i ..< k+i {

                // print("\(maxi) + \(nums[i])")

                maxi = max(nums[j],maxi)

            }

            res.append(maxi)

        }

        return res

    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-6.java
class Solution {
    public int[] productExceptSelf(int[] nums) {

        int product = 0;
        int n = nums.length;

        int[] res = new int[n];

        for(int i=0;i<n;i++){
            product = 1;

            for(int j=0;j<n;j++){

                if(i!=j){
                    product*= nums[j];
                }
            }
            res[i] = product;


        }

        return res;
        
    }
}  
